persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/asset/AssetDAO.kt

```diff
@@ -0,0 +1,50 @@
+package com.wire.kalium.persistence.dao.asset
+
+data class AssetEntity(
+    val key: String,
+    val domain: String,
+    val token: String?,
+    val name: String?,
+    val encryption: String?,
+    val mimeType: String?,
+    val sha: ByteArray?,
+    val size: Long,
+    val downloaded: Boolean
```

Comment:
With the current approach, the downloaded field gets updated when we actually download the asset (public) from the server using the `asset-key` and persist it to the db.

For example:

When we login or open the app, we fetch contacts `SyncContactsUseCase` and each contact have an `asset-key` reference, that we should persist when persisting users `UserRepository.fetchUsersByIds`, at that point I'm persisting only the key with `downloaded` field on false. Other case is when I update my avatar from another client, and we have to download the image on the current device.

How this get downloaded? We, later on demand (ie: usecase fetch, work on PR-243) update the `content`, `size` and `downloaded` field. Also, maybe there could be a worker to download pending assets ? Handle that. This worker that I bring it's not ready, and potentially it isn't a good idea 🙈 . 

It will be helpful if we can do a call to discuss this, as this will be needed the correct approach for the next ticket addressing sending/receiving files https://wearezeta.atlassian.net/browse/AR-933

Reply:
@typfel I'll re-request the review, after we talked earlier this morning, I think we covered that for profile picture this is ok and downloaded field now it makes more sense.

We'll have to address with asset messages, the refactor or extension of this `assets` table or an intermedius one for messages, we'll discuss it tomorrow with Gonzalo again =)